### PR TITLE
refactor: ronny rewritten in golang

### PR DESCRIPTION
**Pull request by NuttyShrimp**

### CODE REVIEW

File: ansible/playbook.yml [hunk 1]
```diff
@@ -45,16 +45,33 @@
...
       ansible.builtin.git:
-        repo: 'https://github.com/12urenloop/Ronny-the-station-chef.git'
-        dest: '/Ronny-the-station-chef'
-        version: 'main'
-    - name: Install specified python requirements
-      pip:
-        requirements: /Ronny-the-station-chef/requirements.txt
-        executable: pip3
-    - name: Ensure myservice  is in a running state
+        repo: 'https://github.com/NuttyShrimp/gonny-the-station-chef.git'
```
Comment (FKD13) [inline ansible/playbook.yml]: This URL should be changed back

File: cmds/emulator/main.go [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+	db := db.New()
+
+	c := make(chan os.Signal, 1)                    // Create channel to signify a signal being sent
+	signal.Notify(c, os.Interrupt, syscall.SIGTERM) // When an interrupt or termination signal is sent, notify the channel
+
+mainLoop:
+	for {
+		select {
+		case <-c:
+			break mainLoop
```
Comment (FKD13) [inline cmds/emulator/main.go]: Is this label needed?

File: cmds/spreader/main.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+			return c.SendString("last_id must be an integer")
+		}
+
+		limit := c.QueryInt("limit", 1000)
+
+		detections, err := conn.GetLimitedIdsAfter(lastId, limit)
+		if err != nil {
+			logrus.Errorf("Failed to retrieve detections from DB: %+v\n", err)
+			return c.SendString("Failed to retrieve detections from DB")
+		}
```
Comment (FKD13) [inline cmds/spreader/main.go]: Error status codes feel appropriate here :D

File: internal/wshandlers/writer.go [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+	// TODO: SQL query
+	if lastId > lastDbId {
+		lastId = lastDbId
+	}
+
+out:
+	for {
+		select {
+		case <-closeChan:
+			break out
```
Comment (FKD13) [inline internal/wshandlers/writer.go]: Same remark here

File: ansible/playbook.yml [hunk 5]
```diff
@@ -5,17 +5,17 @@
...
-        path: /etc/systemd/timesyncd.conf
-        regexp: '^#NTP='
-        line: "NTP=ntp.ugent.be"
-    - name: Enable ntp time sync
-      ansible.builtin.systemd:
-        daemon_reload: true
-        name: systemd-time-wait-sync
-        enabled: true
-        state: started
+    # - name: Set NTP server to the one from UGent
```
Comment (redfast00) [inline ansible/playbook.yml]: It's still important to have synced time; why was this deleted?

File: cmds/collector/main.go [hunk 6]
```diff
@@ -0,0 +1,18 @@
+package main
+
+import (
+	blescanner "github.com/12urenloop/gonny-the-station-chef/internal/ble_scanner"
```
Comment (redfast00) [inline cmds/collector/main.go]: do these URLs need to be changed?

File: cmds/emulator/main.go [hunk 7]
```diff
@@ -0,0 +1,61 @@
...
+
+import (
+	"fmt"
+	"math/rand"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/12urenloop/gonny-the-station-chef/internal/db"
```
Comment (redfast00) [inline cmds/emulator/main.go]: idem

File: cmds/emulator/main.go [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+				time.Sleep(time.Duration(randInt(10, 500)) * time.Millisecond)
+			}
+		}
+	}
+}
+
+func generateRandomDetection() *db.Detection {
+	detection := db.Detection{
+		DetectionTime:     time.Now(),
+		Mac:               fmt.Sprintf("5a:45:55:53:00:%d%d", randInt(0, 10), randInt(0, 10)),
```
Comment (redfast00) [inline cmds/emulator/main.go]: maybe start use 5a:45:55:53:00:a0 up to 5a:45:55:53:00:a9, so we can differentiate random generated data from real data?

File: cmds/spreader/main.go [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+import (
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"strconv"
+	"syscall"
+	"time"
+
+	"github.com/12urenloop/gonny-the-station-chef/internal/db"
```
Comment (redfast00) [inline cmds/spreader/main.go]: idem

File: internal/ble_scanner/scanner.go [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+	"context"
+	"encoding/binary"
+	"fmt"
+	"log"
+	"os/signal"
+	"strings"
+	"syscall"
+	"time"
+
+	"github.com/12urenloop/gonny-the-station-chef/internal/db"
```
Comment (redfast00) [inline internal/ble_scanner/scanner.go]: idem

File: internal/ble_scanner/scanner.go [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+
+	scanner := Scanner{
+		db:  db,
+		ctx: ctx,
+	}
+	return &scanner
+}
+
+func (S *Scanner) scanFilter(a ble.Advertisement) bool {
+	return strings.HasPrefix(a.Addr().String(), ZEUS_MAC_PREFIX)
```
Comment (redfast00) [inline internal/ble_scanner/scanner.go]: Are the MAC addresses guaranteed to be lowercase? I now see that ronny also does this wrong, but maybe we should lowercase the address first before comparing (since we also store lowercase MACs in the DB)

File: ansible/playbook.yml (hunk 5 above)
Comment (NuttyShrimp) [reply ansible/playbook.yml]: My vm kept hanging when using using this, I forgot to uncomment it

Comment (NuttyShrimp) [reply ansible/playbook.yml]: My vm kept hanging when using this, I forgot to uncomment it

**Review by redfast00 — APPROVED** [review]
Thanks for fixing!
